managed/services/agents/mysql.go

```diff
@@ -139,16 +147,33 @@ func mysqldExporterConfig(
 		TemplateLeftDelim:  tdp.Left,
 		TemplateRightDelim: tdp.Right,
 		Args:               args,
-		Env: []string{
+		TextFiles:          textFiles,
+	}
+
+	switch {
+	case !pmmAgentVersion.IsFeatureSupported(version.MysqlExporterMySQL8):
+		env := []string{
 			fmt.Sprintf("DATA_SOURCE_NAME=%s", exporter.DSN(service, models.DSNParams{DialTimeout: time.Second, Database: ""}, nil, pmmAgentVersion)),
 			fmt.Sprintf("HTTP_AUTH=pmm:%s", exporter.GetAgentPassword()),
-		},
-		TextFiles: exporter.Files(),
+		}
+		res.Env = env
+	case textFiles != nil:
```

Comment:
we should create my.cnf file even if tls isn't configured right? Seems like instead of switch case it should be simple if else

Reply:
Yes, my bad. Fixed.